latitudesh/terraform-provider-latitudesh
# feat: API v2023-01-06 support

#### What does this PR do?
Update terraform to comply with our API V3 and do some general adjustments

#### Description of Task to be completed?
- Test resources Creation, Update, Import and Deletion and do the necessary adjustments
- Update Docs

#### How should this be manually tested?
Clone the repository, checkout to this branch before proceeding
- Installation
Before installing you will need to modify the `Makefile` present in the folder with the right information.
Normally you'll only need to modify two fields:
```
VERSION=0.0.1 # Make sure this doesn't interfere with other installations that you might have. I leave it as 0.0.1
OS_ARCH=linux_amd64 # This is your operating system architecture
```
*You can find the full list of supported `OS_ARCH` at `https://github.com/latitudesh/terraform-provider-latitudesh/releases` by expanding Assets.*

Them run `make install` to do a local installation of the provider.

- Setup
Create a new folder with a `main.tf` file:
```
# main.tf
terraform {
  required_providers {
    latitudesh = {
      source  = "latitude.sh/iac/latitudesh"
      version = "~> 0.0.1"
    }
  }
}

# Configure the provider
provider "latitudesh" {
  auth_token = <YOUR_LATITUDESH_TOKEN>
}
```
and run `terraform init`. Now you are ready to start using terraform.

- Testing
We are testing here all of our resources, you can find the general structure that you'll need to use in our `examples` folder. More General instructions can be fount at: https://registry.terraform.io/providers/latitudesh/latitudesh/latest/docs.

After creating your resource files you can run `terraform plan` to verify terraforms action plan. In order to execute it you will need to run 'terraform apply' (in both commands an ssh key will be prompted, for this tests you can leave it blank and proceed)

**IMPORTANT**
After finishing your tests you should run `terraform destroy` to delete all the resources created.

#### Any background context you want to provide?
https://docs.latitude.sh/docs

#### What are the relevant GitHub issues (if any)?
closes https://github.com/latitudesh/terraform-provider-latitudesh/issues/29
